--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1852,6 +1852,7 @@
   configure_package_config_file(CMake/curl-config.cmake.in
           "${project_config}"
           INSTALL_DESTINATION ${CURL_INSTALL_CMAKE_DIR}
+		  PATH_VARS CMAKE_INSTALL_INCLUDEDIR
   )
 
   if(CURL_ENABLE_EXPORT_TARGET)
--- a/CMake/curl-config.cmake.in
+++ b/CMake/curl-config.cmake.in
@@ -38,3 +38,7 @@
 if(NOT TARGET @PROJECT_NAME@::libcurl)
   add_library(@PROJECT_NAME@::libcurl ALIAS @PROJECT_NAME@::@LIB_SELECTED@)
 endif()
+
+# For compatibility with CMake's FindCURL.cmake
+set(CURL_LIBRARIES @PROJECT_NAME@::libcurl)
+set_and_check(CURL_INCLUDE_DIRS "@PACKAGE_CMAKE_INSTALL_INCLUDEDIR@")
